### ISSUE TITLE
records are coupled to test ordering when tests are parameterized

### ISSUE THREAD

**Issue by PeterJCLaw** [adamchainz/django-perf-rec#346]
When using `django-perf-rec` with [`parameterized`](https://pypi.org/project/parameterized/), the records for each test reflect the name of the function as defined in the source, rather than the name of the test. This means that the order in which the tests are run must be stable for the performance records to correctly match their tests.

It would be great if this wasn't the case.

When using `unittest`, it seems there's a `._testMethodName` member on the class instance which contains the current test name (and is used by the `.id()` method, though that returns a fully qualified dotted path to the name, including the module etc.), so using that could be one way to go.

For the moment I'm looking at working around this by passing an explicit `record_name`, but it would be great if this was supported natively.

**Comment by adamchainz** [comment]
Yup that sounds logical. We could call `.id()` and split out the name, or we could access `._testMethodName` - either sounds fine to me.

Pytest functional tests will also have the same issue - perhaps we should access the pytest test ID there.

**Comment by gvangool** [comment]
Our current workaround is this pytest fixture that gets used as the `record_name`
```python
@pytest.fixture
def test_name():
    return os.environ.get("PYTEST_CURRENT_TEST").split(" ")[0].split("::")[-1]
```

Usage afterwards:
```python
@pytest.mark.parametrize("url", ["/", "/admin/"])
def test_pages(client, url, test_name):
    with django_perf_rec.record(record_name=test_name):
        client.get(url)
```

**Comment by adamchainz** [comment]
We could default to using `PYTEST_CURRENT_TEST` like that when defined, would you like to create a PR @gvangool ?

@PeterJCLaw btw I created an alternative to parameterized: https://github.com/adamchainz/unittest-parametrize . That *may* not exhibit the issue, it's a lot simpler...

**Comment by adamchainz** [comment]
Fixed in #537, released in 4.22.3.

**Comment by PeterJCLaw** [comment]
For clarity: should we expect that fix to work for `unittest.TestCase` tests too, or is it just for `pytest`? (I had a quick look at the PR and it seems to be the latter only?)

**Comment by adamchainz** [comment]
You're right, this hasn't been tackled when not using pytest.
